text/0135-template-class-object-binding.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Template class object binding
+status: DRAFTED
+created_at: 2024-01-04
+updated_at: 2024-01-04
+champion: Pierre-Marie Dartus (pmdartus)
+pr: None
+---
+
+# Template class object binding
+
+## Summary
+
+This proposal aims to improve the Developer Experience (DX) around managing components with complex styles by enabling developers to describe the classes applied to an element using JavaScript objects.
+
+## Basic example
+
+```js
+export default class CustomButton extends LightningElement {
+  variant = null;
+  position = null;
+
+  fullWidth = false;
+  stretch = false;
+
+  get className() {
```

Comment:
This property doesn't need to be called `className`, correct? This example may be a bit confusing, since it may look like it's important to override `Element.prototype.className`.